olive/model/handler/diffusers.py

```diff
@@ -0,0 +1,222 @@
+from olive.resource_path import OLIVE_RESOURCE_ANNOTATIONS
+
+if TYPE_CHECKING:
+    from diffusers import DiffusionPipeline
+
+logger = logging.getLogger(__name__)
+
+
+class DiffusersModelType(StrEnumBase):
+    """Diffusion model types."""
+
+    AUTO = "auto"
+    SD15 = "sd15"
+    SDXL = "sdxl"
+    FLUX = "flux"
+
+
+@model_handler_registry("DiffusersModel")
+class DiffusersModelHandler(OliveModelHandler):
+    """Model handler for diffusers models (Stable Diffusion, SDXL, Flux, etc.).
+
+    This handler is designed for diffusion models from the diffusers library.
+
+    Example usage:
+        model = DiffusersModelHandler(
+            model_path="runwayml/stable-diffusion-v1-5",
+            model_type="sd15",  # optional: sd15, sdxl, flux, or auto
+        )
+    """
+
+    resource_keys: tuple[str, ...] = ("model_path", "adapter_path")
+    json_config_keys: tuple[str, ...] = ("model_type", "load_kwargs")
+
+    def __init__(
+        self,
+        model_path: OLIVE_RESOURCE_ANNOTATIONS,
+        model_type: Union[str, DiffusersModelType] = DiffusersModelType.AUTO,
+        load_kwargs: Optional[dict[str, Any]] = None,
+        adapter_path: OLIVE_RESOURCE_ANNOTATIONS = None,
+        model_attributes: Optional[dict[str, Any]] = None,
+    ):
+        """Initialize DiffusersModelHandler.
+
+        Args:
+            model_path: Path to diffusion model (local path or HuggingFace model ID).
+            model_type: Model type: 'sd15', 'sdxl', 'flux', or 'auto' for auto-detection.
+            load_kwargs: Additional kwargs for loading the model (e.g., torch_dtype, variant).
+            adapter_path: Path to LoRA adapter weights.
+            model_attributes: Additional model attributes.
+
+        """
+        super().__init__(
+            framework=Framework.PYTORCH,
+            model_file_format=ModelFileFormat.PYTORCH_ENTIRE_MODEL,
+            model_path=model_path,
+            model_attributes=model_attributes or {},
+        )
+        self.add_resources(locals())
+
+        self.model_type = DiffusersModelType(model_type)
+        self.load_kwargs = load_kwargs or {}
+        self._pipeline = None
+
+    @property
+    def adapter_path(self) -> Optional[str]:
+        """Return the path to the LoRA adapter."""
+        return self.get_resource("adapter_path")
+
+    @property
+    def detected_model_type(self) -> DiffusersModelType:
+        """Detect the diffusion model type."""
+        if self.model_type != DiffusersModelType.AUTO:
+            return self.model_type
+
+        model_path_lower = self.model_path.lower() if self.model_path else ""
+
+        # Check for Flux
+        if "flux" in model_path_lower:
+            return DiffusersModelType.FLUX
+
+        # Try to detect from model config
+        try:
+            from diffusers import FluxTransformer2DModel
+
+            FluxTransformer2DModel.load_config(self.model_path, subfolder="transformer")
+            return DiffusersModelType.FLUX
```

Comment:
Don't we want to update `self.model_type` also here to avoid repeated detection?

Reply:
Makes sense. Upadated!